src/main/scala/inox/tip/TipDebugger.scala

```diff
@@ -47,9 +47,11 @@ trait TipDebugger extends Solver {
     }
   }
 
-  abstract override def assertCnstr(expr: Expr): Unit = {
+  abstract override def assertCnstr(expr: Expr): Unit = try {
     debugOut.foreach { o => o.printScript(encoder.encode(expr)) }
     super.assertCnstr(expr)
+  } catch {
+    case te: program.symbols.TypeErrorException => ctx.reporter.fatalError(te.pos, te.getMessage)
```

Comment:
In the end, I've removed this controversial bit: it's better to let the caller deal with these kind of errors and assume programs typecheck when being solved.